Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  SetupPresenter.swift
+//  Customer Demo
+//
+//  Copyright 2022 Ably Real-time Ltd (ably.com)
+//
+
+import UIKit
+import CoreLocation
+import AblyAssetTrackingSubscriber
+
+class SetupPresenter: NSObject {
+    
+    weak var viewController: SetupViewController?
+    
+    func handleSubmitAction(
+        sourceLocationText: (String, String),
+        destinationLocationText: (String, String),
+        minimumDisplacementText: String,
+        desiredIntervalText: String,
+        accuracyIndex: Int
+    ) {
+        if !validate(
```

Comment:
`guard` here would be probably more readable, but it's such a minor thing that you can feel free to ignore me...